tls/s2n_kem.c

```diff
@@ -249,8 +249,126 @@ int s2n_cipher_suite_to_kem(const uint8_t iana_value[S2N_TLS_CIPHER_SUITE_LEN],
         const struct s2n_iana_to_kem *candidate = &kem_mapping[i];
         if (memcmp(iana_value, candidate->iana_value, S2N_TLS_CIPHER_SUITE_LEN) == 0) {
             *compatible_params = candidate;
-            return 0;
+            return S2N_SUCCESS;
         }
     }
     S2N_ERROR(S2N_ERR_KEM_UNSUPPORTED_PARAMS);
 }
+
+int s2n_get_kem_from_extension_id(struct s2n_blob *kem_id, const struct s2n_kem **kem) {
+    /* cppcheck-suppress knownConditionTrueFalse */
+    S2N_ERROR_IF(kem_mapping == NULL, S2N_ERR_KEM_UNSUPPORTED_PARAMS);
+
+    notnull_check(kem_id);
+    notnull_check(kem_id->data);
+    S2N_ERROR_IF(kem_id->size != 2, S2N_ERR_KEM_UNSUPPORTED_PARAMS);
```

Comment:
Nit: I think passing in a `uint16_t` would be a clearer API contract rather than passing in a `s2n_blob` that must be exactly 2 bytes long.